src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -165,17 +166,20 @@ const JobHistory = _.flow(
                 const {
                   methodConfigurationNamespace, methodConfigurationName, submitter, submissionId
                 } = submissions[rowIndex]
+                const viewInJobManager = linkToJobManager && !!workflowCollectionName
                 return h(Fragment, [
                   div([
                     div([
                       methodConfigurationNamespace !== namespace && span({ style: styles.deemphasized }, [
                         `${methodConfigurationNamespace}/`
                       ]),
-                      link({
+                      link(viewInJobManager ? {
                         target: '_blank',
-                        //check link with actual job manager link
-                        href: linkToJobManager ? `${getConfig().jobManagerUrlRoot}?q=submission-id%3D${submissionId}` :
-                          `${getConfig().firecloudUrlRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
+                        href: `${getConfig().jobManagerUrlRoot}?q=submission-id%3D${submissionId}`
+                      } : {
+                        onClick: () => this.setState({
+                          linkToFC: `${getConfig().firecloudUrlRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
```

Comment:
nit: We could just store the `submissionId` in state, and compute the full URL dynamically.